### ISSUE TITLE
API Limit?

### ISSUE THREAD

**Issue by jkingsman** [keybase/keybase-issues#1804]
Is there a rate limit to be aware of for the API? Of course, I'm going to be as respectful as possible and not make unnecessary requests, but I'm working on some friend-finding systems that would be making lots of repeated queries, and I wanted to know if there's any rate limiting or total request limiting I should abide by, please.

Thanks!

**Comment by oconnor663** [comment]
There is. Our nginx server has a `limit_req` config of `burst=50 nodelay`. I don't know exactly what behavior that entails though. (And tell me if it's bad practice for me to publish that. I assume it's not secret insofar as you can figure it out by just hitting the limit.)

**Comment by maxtaco** [comment]
Most sites with APIs publish this info, we should do the same...

**Comment by jkingsman** [comment]
I believe `burst=50` means that requests will be processed at the defined rate (look for `rate = x r/s`, where x is a number) for a maximum of 50 consecutive requests from an IP before a 503 is returned. As I understand it, the 503 is only returned if requests are being handled at rate capacity; if they're staggered at less than the defined `r/s`, the `burst` config is irrelevant.

**Comment by oconnor663** [comment]
Our config is similar to the last example at http://nginx.org/en/docs/http/ngx_http_limit_req_module.html#limit_req. It doesn't explicitly specify `rate`.

**Comment by PeaStew** [comment]
Well I do seem to have hit a different limit 
```GetConversationForCLILocal error: rate limit violation: limit [chat, 15m0s, 9000] calls: 9000 remaining: 0s```

**Comment by mmaxim** [comment]
@PeaStew can you send a log, your client seems to be in some kind of weird state.

**Comment by PeaStew** [comment]
@mmaxim and the log would be where on Debian? Can't find it amongst the usual suspect locations

**Comment by maxtaco** [comment]
`keybase log send`

Logs are in `~/.cache` I believe

**Comment by PeaStew** [comment]
7101090cb9de316960e8b01c

**Comment by mmaxim** [comment]
@PeaStew we have a fix that will land today you all can use. Is there any reason you all aren't using the chat JSON API? It is available from `keybase chat api --help`

**Comment by PeaStew** [comment]
@mmaxim Arghh you are a life saver, the help docs don't show anything but this was exactly what I was looking for :)

**Comment by mmaxim** [comment]
Yeah sorry our docs around it are not in a great state. The good news is right now there is an effort to make it a lot better, hopefully will land soon. Glad it helps!
